src/services/room-service.js

```diff
@@ -22,12 +22,48 @@ export default class RoomService {
     this.userService = userService;
   }
 
+  async getRooms({ ownerId, memberId, access }) {
+    const andFilters = [];
+    const orFilters = [];
+
+    if (ownerId) {
+      orFilters.push({ owner: ownerId });
+    }
+    if (memberId) {
+      orFilters.push({ members: { $elemMatch: { userId: memberId } } });
+    }
+
+    if (orFilters.length === 1) {
+      andFilters.push(orFilters[0]);
+    }
+    if (orFilters.length > 1) {
+      andFilters.push({ $or: orFilters });
+    }
+
+    if (access) {
+      andFilters.push({ access });
+    }
+
+    let filter = {};
+    if (andFilters.length === 1) {
+      filter = andFilters[0];
+    }
+    if (andFilters.length > 1) {
+      filter = { $and: andFilters };
+    }
+
+    const rooms = await this.roomStore.find(filter);
+    return rooms;
+  }
```

Comment:
Since adding this method initially, I ended up rethinking the logic and not needing `andFilters`.
However, I kept this method as is, since I thought it's a good base setup for the future and has proper tests.

What's your take on it? Should we reduce it to only have `{ ownerId, memberId }` logic, as `access` is not used so far?

Reply:
You can leave it, but it seems a bit involved int his way. I think we can write this more easily, we can take a look if you want.

Reply:
"but it seems a bit involved int his way." what do you mean by this?

Reply:
I think this should work in the same way:
```
const ownerFilter = {
      owner: ownerId,
      ...access && { access }
    };

    const memberFilter = {
      'members.userId': memberId,
      ...access && { access }
    };

    if (ownerId) {
      orFilters.push(ownerFilter);
    }
    if (memberId) {
      orFilters.push(memberFilter);
    }

    const filter = orFilters.length ? orFilters : {};

    const rooms = await this.roomStore.find(filter);
```

Reply:
If you change the code and place yours instead, 6 tests fail 😬 

I think I might have not made it clear. `access` is not passed in with the current usage, it's `undefined`. So can all the others. I wanted to make a generic method that can be scaled up and can support lacking of any of the arguments.

Reply:
But this is as generic, maybe you need to spread the  access also on the stand alone filter. 
The tests fail because they expect the filter to look in a certain way instead of actually querying the db or?
I'm currently checking with the code on local.

Reply:
It works the same with your db setup and this code on local :).
```
 async getRooms({ ownerId, memberId, access }) {
    const orFilters = [];
    const ownerFilter = {
      owner: ownerId,
      ...access && { access }
    };

    const memberFilter = {
      'members.userId': memberId,
      ...access && { access }
    };

    if (ownerId) {
      orFilters.push(ownerFilter);
    }
    if (memberId) {
      orFilters.push(memberFilter);
    }

    const filter = orFilters.length ? orFilters : { ...access && { access } };

    const rooms = await this.roomStore.find(filter);
    return rooms;
  }
```

Reply:
Some fail because the Mongo query is invalid.
In your code you combine `access` into the other 2 filters, but that's not what is needed. For example, you could use this method to simply `getRooms({ access: 'private' })`. With the current code, this works, with your change it doesn't.
This is what I mean by generic.

Reply:
@ahelmberger, @laurentiu-ilici, I reduced `getRooms` how we discussed, please take a look.

Reply:
:+1: Thanks!